File: Typescript/Practice/26072022/medium/medium_addingSuffixes.py
#!/usr/bin/env python

def add_suffix(suffix):
  def func(string):
    return string + suffix
  return func

add_ly = add_suffix("ly")

print(add_ly("hopeless"))
print(add_ly("total"))

add_less = add_suffix("less")

print(add_less("fear"))
print(add_less("ruth"))

File: Typescript/Practice/medium/medium_addingSuffixes.ts
#!/usr/bin/env npx ts-node

function add_suffix (suffix: string) {
  let func = function (str: string): string {
    return str + suffix;
  };
  return func
};

const add_ly = add_suffix("ly");

console.log(add_ly("hopeless"));
console.log(add_ly("total"));

const add_less = add_suffix("less");

console.log(add_less("fear"));
console.log(add_less("ruth"));
